### DISCUSSION TITLE
Add custom context

### DISCUSSION THREAD

**Discussion by ilyalasy** [facebookresearch/ParlAI#4154] [Q&A]
Hey, sorry for probably silly question:
How can I finetune blenderbot on bst dataset but with custom context? I'd like to append knowledge text (already retrieved) to each dialog utterance. Should I implement custom Teacher?
Thanks!

**Comment by mojtaba-komeili** [comment]
I suggest you check out the [Mutators](https://parl.ai/docs/mutators.html). They allow you to modify any dataset (Teacher). For the case that you mentioned, [this](https://github.com/facebookresearch/ParlAI/blob/825a05729061787b45224d7df4131be036ed0754/parlai/tasks/wizard_of_wikipedia/agents.py#L1297) might be something fairly similar. You can use as a starting point to make your own mutator.

**Reply by ilyalasy** [reply]
Thanks, I'll check this out

**Reply by stephenroller** [reply]
You might also be interested in temp_history
